modules/gmc-agent/src/main/java/org/gridgain/action/controller/QueryActionsController.java

```diff
@@ -92,8 +97,10 @@ public QueryActionsController(GridKernalContext ctx) {
                 String qryId = requireNonNull(arg.getQueryId(), "Failed to execute query due to empty query ID");
```

Comment:
We should check all requirements before submitting to pool, to reduce context switch

Reply:
Agree.